common/lib/plugins/custom_endpoint/custom_endpoint_monitor_impl.ts

```
@@ -0,0 +1,178 @@
+  protected static customEndpointInfoCache: CacheMap<string, CustomEndpointInfo> = new CacheMap();
+
+  private rdsClient: RDSClient;
+  private customEndpointHostInfo: HostInfo;
+  private endpointIdentifier: string;
+  private readonly region: string;
+  private readonly refreshRateMs: number;
+
+  private pluginService: PluginService;
+  private infoChangedCounter: TelemetryCounter;
+
+  protected stop = false;
+  private timerA: any = {};
+  private timerB: any = {};
+  private timerC: any = {};
+
+  constructor(
+    pluginService: PluginService,
+    customEndpointHostInfo: HostInfo,
+    endpointIdentifier: string,
+    region: string,
+    refreshRateMs: number,
+    rdsClientFunc: (hostInfo: HostInfo, region: string) => RDSClient
+  ) {
+    this.pluginService = pluginService;
+    this.customEndpointHostInfo = customEndpointHostInfo;
+    this.endpointIdentifier = endpointIdentifier;
+    this.region = region;
+    this.refreshRateMs = refreshRateMs;
+    this.rdsClient = rdsClientFunc(customEndpointHostInfo, this.region);
+
+    const telemetryFactory = this.pluginService.getTelemetryFactory();
+    this.infoChangedCounter = telemetryFactory.createCounter(CustomEndpointMonitorImpl.TELEMETRY_ENDPOINT_INFO_CHANGED);
+
+    this.run();
+  }
+
+  async run(): Promise<void> {
+    logger.verbose(Messages.get("CustomEndpointMonitorImpl.startingMonitor", this.customEndpointHostInfo.host));
+
+    while (!this.stop) {
+      try {
+        const start = Date.now();
+        this.timerB = null;
+
+        const input = {
+          DBClusterEndpointIdentifier: this.endpointIdentifier,
+          Filters: [
+            {
+              Name: "db-cluster-endpoint-type",
+              Values: ["custom"]
+            }
+          ]
+        };
+        const command = new DescribeDBClusterEndpointsCommand(input);
+        const result = await this.rdsClient.send(command);
+
+        const endpoints = result.DBClusterEndpoints;
+        if (endpoints.length !== 1) {
```

Can we check and fail early if `endpoints.length === 0`?